canonical-web-and-design/maas-ui
# feat(ui): Build RSD list in React

## Done

- The RSD components have mostly been copied over from the KVM components, with minor changes where necessary. I originally tried to just have a single component `Pod` that could be given a prop `type="rsd"` or `type="kvm"` but the amount of inline conditionals in the component got pretty out of hand. So I think this is a cleaner option that can be refactored after the fact if necessary.
- For now the "Add RSD" button takes you to the form on the machine list but I'll create a component for that next.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Get a MAAS running with RSDs. Using `make sampledata` creates 3 KVMs and 3 RSDs.
- Check that RSD shows up in the main navigation.
- Go to /r/rsd manually (I'll update the nav link once it's all complete) and check that the RSDs show up correctly.
- Select some and check that the checkboxes and the header text respond accordingly
- Check that deleting works
- Check that refreshing sends the correct websocket response, but times out because it's not actually real.

## Fixes

Fixes #1638 

## Screenshot

![0 0 0 0_8400_MAAS_r_rsd (2)](https://user-images.githubusercontent.com/25733845/93729503-17f39a00-fc08-11ea-8e3f-a3dc1e4d271c.png)
